cmake/scripts/buildinfo.cmake

```diff
@@ -19,20 +19,19 @@ if (NOT ETH_BUILD_PLATFORM)
 	set(ETH_BUILD_PLATFORM "unknown")
 endif()
 
-# Logic here: If prerelease.txt exists but is empty, it is a non-pre release.
-# If it does not exist, create our own prerelease string
-if (EXISTS ${ETH_SOURCE_DIR}/prerelease.txt)
-	file(READ ${ETH_SOURCE_DIR}/prerelease.txt SOL_VERSION_PRERELEASE)
-	string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
-else()
+# Logic here: If prerelease.txt has the templated placeholder, it is a pre release.
+file(READ ${ETH_SOURCE_DIR}/prerelease.txt SOL_VERSION_PRERELEASE)
+string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
+If (SOL_VERSION_PRERELEASE MATCHES ^[$]Format:)
```

Comment:
That's not the same as had happened before, is it?
If we are in a source archive, then ``prerelease.txt`` will not contain the placeholder, but it *will* contain whatever ``$Format:%d$`` resolves to - which won't be "empty". So releases will become prereleases... won't they?